src/main/java/frc/lib/io/motor/MotorIORev.java

```
@@ -0,0 +1,236 @@
+import com.revrobotics.spark.ClosedLoopSlot;
+import com.revrobotics.spark.SparkBase;
+import com.revrobotics.spark.SparkBase.PersistMode;
+import com.revrobotics.spark.SparkBase.ResetMode;
+import com.revrobotics.spark.SparkClosedLoopController;
+import com.revrobotics.spark.SparkFlex;
+import com.revrobotics.spark.SparkMax;
+import com.revrobotics.spark.config.SparkBaseConfig;
+import com.revrobotics.spark.config.SparkBaseConfig.IdleMode;
+
+import com.revrobotics.spark.config.SparkFlexConfig;
+import com.revrobotics.spark.config.SparkMaxConfig;
+import edu.wpi.first.units.measure.Angle;
+import edu.wpi.first.units.measure.AngularVelocity;
+import edu.wpi.first.units.measure.Current;
+import edu.wpi.first.units.measure.Voltage;
+import frc.lib.util.Device;
+import com.revrobotics.spark.SparkLowLevel.MotorType;
+
+public class MotorIORev implements MotorIO {
+
+    public SparkBase motor;
+    public SparkClosedLoopController controller;
+    private RelativeEncoder encoder;
+
+    private final String name;
+
+    /**
+     * REV WORKAROUNDS
+     * 
+     * The motorIO interface was made for CTRE hardware. Some code does not map 1:1 to REV. The
+     * below are workarounds to make the mapping easier.
+     */
+
+    /**
+     * Maps MotorIO.PIDSlot to REV ClosedLoopSlot.
+     */
+    private static final EnumMap<MotorIO.PIDSlot, ClosedLoopSlot> SLOT_MAP =
+        new EnumMap<>(MotorIO.PIDSlot.class);
+    static {
+        SLOT_MAP.put(MotorIO.PIDSlot.SLOT_0, ClosedLoopSlot.kSlot0);
+        SLOT_MAP.put(MotorIO.PIDSlot.SLOT_1, ClosedLoopSlot.kSlot1);
+        SLOT_MAP.put(MotorIO.PIDSlot.SLOT_2, ClosedLoopSlot.kSlot2);
+    }
+
+    /**
+     * Gets the corresponding REV ClosedLoopSlot for a given MotorIO.PIDSlot.
+     * 
+     * @param slot The MotorIO.PIDSlot to convert.
+     */
+    protected ClosedLoopSlot getClosedLoopSlot(MotorIO.PIDSlot slot)
+    {
+        return SLOT_MAP.getOrDefault(slot, ClosedLoopSlot.kSlot0);
+    }
```

These enums should map 1-1, this should throw an `IllegalStateException` rather than silently defaulting